sympy/solvers/tests/test_solveset.py

```diff
@@ -3271,3 +3271,13 @@ def test_issue_22628():
 
 def test_issue_25781():
     assert solve(sqrt(x/2) - x) == [0, S.Half]
+
+
+def test_issue_25423():
+    x, y = symbols('x y')
+    equations1 = []
+    equations2 = [2 * x + 3 * y - 1]
+    nonlinear_equations = [x ** 2 + y ** 2 - 1, x - y]
+    raises(TypeError, lambda: linear_eq_to_matrix(equations1, {x, y}))
+    raises(TypeError, lambda: linear_eq_to_matrix(equations2, {x, y}))
+    raises(NonlinearError, lambda: linear_eq_to_matrix(nonlinear_equations, x, y))
```

Comment:
Instead of this, how about showing/affirming that passing `set(equations2)` with `[x, y]` does *not* raise an error.

Also, symbols `x` and `y` have already been created at the top of the file and need not be recreated here.

Reply:
In the case of `set (equations2)`, there will be test failure indicates that the current implementation of `linear_eq_to_matrix` does not support a set of equations as input. It will raise exception becasue Python set is not recognized as a valid sequence for this purpose. 

In this case, for the test to work, I will need to convert `sets` to `list`. In such a case, we are not testing the function's ability to handle sets directly. It's a gap in our testing if the function should handle sets.

Let me know what you think :)

Reply:
OK, that's fine (and probably better) that an error raises. It won't change the solution but it could be a little harder to debug if the order changes wrt what was passed. Then let's delete that 3rd test since it is already tested in lines 1408ff in the same file.